File: app/src/main/java/ownmanager/in/fragmentruntimesample/fragments/FirstFragment.java
package ownmanager.in.fragmentruntimesample.fragments;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;

import androidx.fragment.app.Fragment;

import ownmanager.in.fragmentruntimesample.R;


public class FirstFragment extends Fragment {
    private Button secondFragment;
    public FirstFragment() {
        // Required empty public constructor
    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        // Inflate the layout for this fragment
        View view = inflater.inflate(R.layout.fragment_first, container, false);

        secondFragment = view.findViewById(R.id.secondFragmentBtn);

        // Moving from First Fragment to Second Fragment
        secondFragment.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                getActivity().getSupportFragmentManager() // replaces second fragment with main activity container
                        .beginTransaction()
                        .replace(R.id.fragment_container,new FragmentToActivityCommunication(),null)
                        .addToBackStack(null) // instead of destroy it will stop the fragment -> makes navigation between fragments possible
                        .commit();
            }
        });
        return view;
    }

}

File: app/src/main/java/ownmanager/in/fragmentruntimesample/fragments/nonImportant/FirstFragment.kt
package ownmanager.`in`.fragmentruntimesample.fragments.nonImportant

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import kotlinx.android.synthetic.main.fragment_first.view.*
import ownmanager.`in`.fragmentruntimesample.R
import ownmanager.`in`.fragmentruntimesample.fragments.FragmentToActivityCommunication

class FirstFragment : Fragment() {
    override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        // Inflate the layout for this fragment
        val view: View = inflater.inflate(R.layout.fragment_first, container, false)

        // Moving from First Fragment to Second Fragment
        view.secondFragmentBtn.setOnClickListener(View.OnClickListener {
            activity!!.supportFragmentManager // replaces second fragment with main activity container
                .beginTransaction()
                .replace(
                    R.id.fragmentContainer,
                    FragmentToActivityCommunication(), null
                )
                .addToBackStack(null) // instead of destroy it will stop the fragment -> makes navigation between fragments possible
                .commit()
        })
        return view
    }
}